client/src/ui/molecules/offline-status-bar/index.tsx

```diff
@@ -0,0 +1,51 @@
+import * as React from "react";
+
+import { isPayingSubscriber } from "../../../utils";
+import { useUserData } from "../../../user-context";
+
+import "./index.scss";
+
+export function OfflineStatusBar() {
+  const user = useUserData();
+  const statusBarRef = React.useRef<HTMLDivElement>(null);
+  const initialOnlineStatus =
+    navigator && "onLine" in navigator ? navigator.onLine : true;
+  const [isOnline, setIsOnline] = React.useState(initialOnlineStatus);
```

Comment:
You can reuse `useOnlineStatus()` from `client/src/hooks.ts`, and refine it with the `navigator && "onLine" in navigator` check.

Reply:
Switched to the hook.